src/swap/calculateEthForExactNxm.ts

```diff
@@ -0,0 +1,24 @@
+import { Reserves } from './reserves.type';
+
+/**
+ * Calculates the amount of eth needed to flow into the pool to get the exact amount of nxm out
+ */
+// suggestion: calculateEthInGivenNxmOut
+export const calculateEthForExactNxm = (nxmOut: bigint, reserves: Reserves): bigint => {
+
+  if(nxmOut <= 0n) throw new Error('NXM out value must be greater than 0');
```

Comment:
I think `if(nxmOut > reserves.nxmA) { ... }` should be there as well and would prevent the negative `ethIn`

Reply:
I assume for the other exact method would be `ethOut > reserves.ethReserve`. I'll add the conditions for the others as well 👍